### PR TITLE
feat: add Markdown linting to Gruntfile (fix #38)

### LINKED ISSUE #38: markdown files are not being linted
## Describe the bug

Running `npm run lint` does not lint the markdown files because the [`lintall`](https://github.com/fluid-project/docs-inclusive-learning/blob/0d32acda319f6a2fc1c6a69a8357f8684143cb86/Gruntfile.js#L10-L14) task has not been configured to locate them.

### To reproduce

1. From the terminal run `npm run lint`
2. Notice that the markdown files have not been linted

### Expected behavior

The markdown files should have been linted, and produce no errors.

### PR DESCRIPTION
**Pull request by greatislander**
Also: fix Markdown linting issues, add CI for linting.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -10,7 +10,8 @@ module.exports = function (grunt) {
         lintAll: {
             sources: {
                 json: ["./*.json", ".eslintrc.json", ".stylelintrc.json", "./src/**/*.json"],
-                js: ["./*.js", ".eleventy.js", "src/**/*.js", "!src/lib/**/*.js"]
+                js: ["./*.js", ".eleventy.js", "src/**/*.js", "!src/lib/**/*.js"],
+                md: ["./src/documents/*.md","README.md","CONTRIBUTING.md","AUTHORS.md","./src/404.md", "!dist/**/*.md"]
```
Comment (jobara) [inline Gruntfile.js:14]: Please add a space after the "," for legibility.

File: README.md [hunk 2]
```diff
@@ -32,10 +32,13 @@ If you make changes to the website, repeat the steps to build the image and star
...
 
-If you make changes to the website, repeat step 2 to build the website and upload any changed files from the `./dist/` directory to the web root of your server.
+If you make changes to the website, repeat step 2 to build the website and upload any changed files from the `./dist/`
+directory to the web root of your server.
 
 ## License Information
 
-The Inclusive Learning Design Handbook's code is licensed under the [BSD 3-Clause](https://raw.githubusercontent.com/fluid-project/docs-inclusive-learning/blob/main/LICENSE.md) license.
+The Inclusive Learning Design Handbook's code is licensed under the
+[BSD 3-Clause](https://raw.githubusercontent.com/fluid-project/docs-inclusive-learning/blob/main/LICENSE.md) license.
```
Comment (jobara) [inline README.md:41]: The link to the license file is broken.
```suggestion
[BSD 3-Clause](https://raw.githubusercontent.com/fluid-project/docs-inclusive-learning/main/LICENSE.md) license.
```

File: src/documents/CaseStudyExemplar.md [hunk 3]
```diff
@@ -72,6 +83,7 @@ In order to use scripts in EPUB, CDATA blocks need to be added. However this wil
 </script>
 ```
 
+<!-- markdownlint-disable ol-prefix -->
```
Comment (jobara) [inline src/documents/CaseStudyExemplar.md:86]: According to https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md029---ordered-list-item-prefix the rules should still pass if the code blocks are indented. Could you see if you can make it work without disabling the rule itself?

File: src/documents/EPUB3Resources.md [hunk 4]
```diff
@@ -3,34 +3,57 @@ title: EPUB 3 Resources
...
+- [WAI-ARIA](/WAI-ARIA.html)
+- [Visual Styles](/VisualStyles.html)
+- [Graphics and Multimedia](/GraphicsAndMultimedia.html)
+- [Scripting in EPUB 3](/ScriptingInEPUB3.html)
+- [Mathematics](/Mathematics.html)
+- [Narrations and Text to Speech](/NarrationsAndTextToSpeech.html)
+- [Accessibility Metadata](/AccessibilityMetadata.html)
+- [Testing and Validation](/TestingAndValidation.html)
+- [Case Study Exemplar](/CaseStudyExemplar.html)
+- [EPUB 3 Resources](/EPUB3Resources.html)
```
Comment (jobara) [inline src/documents/EPUB3Resources.md:59]: Why the change from "*" to "-"? It seems that we use "*" in other lists throughout.

Comment (greatislander) [reply src/documents/EPUB3Resources.md:59]: For some reason the linter was flagging this one. I'm not entirely sure.

File: src/documents/StandardizedTestTools.md [hunk 5]
```diff
@@ -90,30 +112,37 @@ This item requires  examination on a given topic requiring a written analysis or
...
 </ul>
 </td>
 </tr>
 <tr>
-<td>** *Labeling Questions* **,
-** *Grouping and Ordering* **,
-** *Interactive Questions* **
+<td>***Labeling Questions***,
+***Grouping and Ordering***,
+***Interactive Questions***
```
Comment (jobara) [inline src/documents/StandardizedTestTools.md:145]: There is something wrong with basically everything in this column. I think what the intention may have been was to have things both **bold** and _italicized_. However, they are inside of HTML markup, so we'd have to actually do this with HTML and not Markdown. Maybe these could be `<th>` but not sure if that makes sense in this context.

File: src/documents/StandardizedTestTypes.md [hunk 6]
```diff
@@ -5,9 +5,14 @@ category: Approaches
...
+Low-stakes testing is a form of assessment encompassed by the immediate process of learning, often in a very short
+feedback loop, such as exercises or quizzes. Sometimes this is called "formative" standardized testing or even just
+"feedback". The essential characteristics are immediacy and the lack of serious consequences contingent on performance.
 
-**formative** **tests** are generally considered as low-stakes tests that have low or no point value. The goal of a formative test is to monitor student learning to provide ongoing feedback that can be used by instructors to recognize where students are struggling and address problems immediately and by students to identify their strengths and weaknesses and target areas that need work. Examples of formative tests include:
+**formative** **tests** are generally considered as low-stakes tests that have low or no point value. The goal of a
```
Comment (jobara) [inline src/documents/StandardizedTestTypes.md:12]: This should probably be written as `**formative tests**`

File: src/documents/StandardizedTestTypes.md [hunk 7]
```diff
@@ -24,13 +29,19 @@ Low-stakes tests and formative tests often take place in one of the following co
...
 
 ## High-stakes Test
 
-High-stakes testing has consequences that may make a serious impact on the life-course of the participant. An example might be a university entrance examination or high school exit exams. It is important that a high-stakes standardized testing be fair to all candidates and not offer advantages to one group over another.
-**summative** **tests** are considered as high-stakes tests and have a high point value. The goal of summative tests is to evaluate student learning at the end of an instructional unit by comparing it against some standard or benchmark.  Information from summative standardized testings can be used formatively when students or faculty use it to guide their efforts and activities in subsequent courses. Examples of summative tests include:
+High-stakes testing has consequences that may make a serious impact on the life-course of the participant. An example
+might be a university entrance examination or high school exit exams. It is important that a high-stakes standardized
+testing be fair to all candidates and not offer advantages to one group over another. **summative** **tests** are
```
Comment (jobara) [inline src/documents/StandardizedTestTypes.md:40]: should this be `**summative tests**`?

File: src/documents/WebGamesAndSimulations.md [hunk 8]
```diff
@@ -36,33 +44,53 @@ If presenting learners with multiple choices that affect the outcome of a game:
...
 ## Content Accessibility / Screen Readers
 
 Follow a good structure and use semantic markup:
 
 * [WebAim.org - Designing for Screen Reader Compatibility](http://webaim.org/techniques/screenreader/)
 * [Inclusive Learning Design Handbook - Accessibility Principles](FollowAccessibilityPrinciples.html)
-* Make sure to use HTML 5 semantic markup wherever possible: i.e. `[https://developer.mozilla.org/en/docs/Web/HTML/Element](https://developer.mozilla.org/en/docs/Web/HTML/Element)
-* Use WAI-ARIA to improve accessibility. See: [Mozilla Developer ARIA - good resource for learning, examples, blogs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA).
+* Make sure to use HTML 5 semantic markup wherever possible: i.e.
+  `[https://developer.mozilla.org/en/docs/Web/HTML/Element](https://developer.mozilla.org/en/docs/Web/HTML/Element)
```
Comment (jobara) [inline src/documents/WebGamesAndSimulations.md:80]: unnecessary "`" at the beginning.

File: src/documents/AchievingInclusiveEducation.md [hunk 9]
```diff
@@ -37,7 +48,8 @@ Learning needs that affect learning can include:
 <figure>
 [![Floe concept design mindmap](/assets/images/thumbs/400px-Floe_concept_design_mindmap.png)](/assets/images/Floe_concept_design_mindmap.png)
 <figcaption>
-*Figure 1:* Floe concept design mindmap illustrates the many different ideas and parts which work together to enable a one-size-fits-one education.
+*Figure 1:* Floe concept design mindmap illustrates the many different ideas and parts which work together to enable a
```
Comment (jhung) [inline src/documents/AchievingInclusiveEducation.md:51]: An unecessary `<br/>` appears at the end of this line when rendered, causing the next line to always be on a separate line.

image

Comment (greatislander) [reply src/documents/AchievingInclusiveEducation.md:51]: Good catch. I guess we need to exempt markdown within figures from the line length rule?

File: src/documents/StandardizedTestTools.md (hunk 5 above)
Comment (greatislander) [reply src/documents/StandardizedTestTools.md:145]: Addressed in #40.

File: src/documents/AchievingInclusiveEducation.md (hunk 9 above)
Comment (jhung) [reply src/documents/AchievingInclusiveEducation.md]: Yes I think that makes sense.

File: src/documents/MeetLearnerNeedsAndPreferences.md [hunk 10]
```diff
@@ -43,34 +55,52 @@ Examples of supplementation include:
...
+translate a symphony orchestra performance into a non-aural experience. In these cases, a revisit of the content’s
+intent and meaning is needed to provide an interpretation of the content into a different medium.
 
 Examples of recreation include:
 
-* For a graphic novel, provide a written novel “equivalent”, based on an interpretation of a combination of the visuals and dialogue.
+* For a graphic novel, provide a written novel “equivalent”, based on an interpretation of a combination of the visuals
+  and dialogue.
 
 <a name="TheImportanceOfNaturalLanguage"></a>
```
Comment (jhung) [inline src/documents/MeetLearnerNeedsAndPreferences.md:73]: Named anchors are having the `link-external` class applied.

image

File: src/documents/EPUB3Resources.md (hunk 4 above)
Comment (greatislander) [reply src/documents/EPUB3Resources.md]: Fun with linting:

1. The default bullet style for ordered lists is `consistent`, aka either `*` or `-` as long as you are consistent within the same file.
2. The ordered list of resources at the beginning of this file uses hyphens to split up content, e.g.:
    > EPUB3 Overview - www.idpf.org/epub/30/spec/epub30-overview.html - answers basic questions about what is EPUB, what it is capable of, and how it can be used.
3. Wrapping to 120 columns ended up placing some of those hyphens at the start of a line, making MarkdownLint (and Markdown) interpret them as nested ordered lists.
4. Now we are failing the consistency rule by using `*` for bullets on the real unordered list at the bottom of the file.

I resolved this by using em dashes instead of hyphens above.

File: src/documents/MeetLearnerNeedsAndPreferences.md (hunk 10 above)
Comment (greatislander) [reply src/documents/MeetLearnerNeedsAndPreferences.md:73]: @jhung Can you open a ticket for this? It'll need to be addressed in guide.inclusivedesign.ca as well. Thanks.

File: src/documents/AchievingInclusiveEducation.md (hunk 9 above)
Comment (jobara) [reply src/documents/AchievingInclusiveEducation.md]: I'm not sure what the issue is exactly, but depending on the parser you may not be able to mix the markdown inside of the HTML.

**Review by jobara — APPROVED** [review]

**Comment by jobara** [comment]
@jhung I'm finished my passthrough, leaving for you, in case you wanted any more changes.

File: src/documents/AchievingInclusiveEducation.md (hunk 9 above)
Comment (greatislander) [reply src/documents/AchievingInclusiveEducation.md]: @jobara See notes about figures here: https://github.com/fluid-project/docs-inclusive-learning/pull/29

File: src/documents/MeetLearnerNeedsAndPreferences.md (hunk 10 above)
Comment (greatislander) [reply src/documents/MeetLearnerNeedsAndPreferences.md:73]: Resolved in 336a17d.
